substrate/frame/identity/src/simple.rs

```diff
@@ -0,0 +1,185 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use enumflags2::{bitflags, BitFlags};
+use frame_support::{traits::Get, CloneNoBound, EqNoBound, PartialEqNoBound, RuntimeDebugNoBound};
+use scale_info::{build::Variants, Path, Type, TypeInfo};
+use sp_runtime::{BoundedVec, RuntimeDebug};
+use sp_std::prelude::*;
+
+use crate::types::{Data, IdentityFields, IdentityInformationProvider, U64BitFlag};
+
```

Comment:
This is just the legacy stuff for out chain?  
Maybe calling it `legacy` or do you plan on maintaining it after the transition?  
Otherwise people will start using it downstream and then we have to make breaking changes to it.  
Could also be guarded by the `experimental` flag.

Reply:
I think calling it `legacy` instead of `simple` works much better. I didn't know what to name it initially. I think it's fine if people use it downstream until we have the new version merged, the one discussed below. At the time of this PR it is in fact the current `struct` used in all chains. I'm indifferent to the `experimental` flag option.

Reply:
`experimental` shouldn't be used here as this is not "experimental". I would also have gone with `legacy`, because the indetity is not really simple as it contains quite a lot of fields.